accepted/0000-replace-buildpack-yml.md

```diff
@@ -0,0 +1,52 @@
+
+## Motivation
+
+With this change we can completely eliminate `buildpack.yml` which is a file that adds undue complexity to the buildpacks because in order for `buildpack.yml` influence what appears in the application image it must be parsed during detection in every buildpack. `buildpack.yml` also has a non-standard format meaning that every single buildpack needs to have slightly unique parsing logic in order to grab a buildpack specific key name and then any other additional metadata.
+
+The big advantage of moving to using `build-plan` is that every buildpack that currently has a `buildpack.yml` parser could remove that code entirely and switch to using Buildpack Plan metadata during the `build` phase and there would be no loss of functionality. The job of parsing build configuration will be condensed into the `build-plan` and then all subsequent parsing will be handled by the buildpack library or the lifecycle.
+
+The format used in `build-plan` is also more powerful than `buildpack.yml` because it allows for fine grain controls as to when that dependency is available by setting `build` and `launch` flags. Any other data that is not the dependency name or version can be put in the metadata section of requires and handled during the `build` phase.
+
+By using the `build-plan` buildpack we can completely replace the `buildpack.yml` while losing no functionality, increase user configurability with very low overhead cost for buildpack authors, and vastly increased interoperability.
+
+The ideal end state for this track or work would be to bake this functionality directly into the lifecycle because of how closely the `build-plan` buildpack already interacts with the lifecycle. The `build-plan` buildpack is a middle ground to test out whether or not this model is a good thing for users and buildpack authors. Once this model has proven (or disproven) itself, learnings from this stop gap should be used to inform further actions (whether that be pursuing integration in the lifecycle or abandoning this model for something else).
+
+It is worth mentioning that the `build-plan` buildpack will be able to `require` any other buildpack, but if that buildpack is not in the group that the `build-plan` buildpack is running in then detection will fail. The `build-plan` buildpack needs to be running in a group with all of its required buildpacks which is something that can be configured by an operator or someone with operator level permissions.
+
+## Implementation
+
+The `build-plan` buildpack would need to be added to the end of every build order. If you would like a specific dependency then you can specify that is a `plan.toml`, which is the `requires` portion of a Build Plan (TOML).
```

Comment:
A better syntax be to be the _full_ `buildplan.toml` hierarchy, including `[or]`, in which all keys other than `[requires]` and it's children are ignored.  Best to properly mimic the thing you are duck-typing to ensure you get the full functionality and ease of future upgrades.

Reply:
I am fine with implementing the `[or]`structure into `plan.toml` but I feel that it would be inappropriate to include the `[provides]` section of `buildplan.toml` as the provisions should be given by the functional implementation buildpacks.

Reply:
You can achieve this by following [Postel's Law](https://en.wikipedia.org/wiki/Robustness_principle).  If you have a strict subset of the build plan syntax, ignoring what you don't care about `[provides]` then you get the most compatible option without expanding the types of documents we already have.

Reply:
I'm not sure how to interpret that comment. 
1. If you a purposing also adding `[provides]` I would argue that having a field in configuration that is ignored during execution feels like bad user experience.
1. If you are purposing I make the `plan.toml` look like `buildplan.toml` except for `[provides]` then I would be all for that and I would happily make the necessary changes to the `build-plan` buildpack

Reply:
The latter precisely.

Reply:
Changes are being made to the `build-plan` buildpack as we speak, I will update the RFC when those are landed.

Reply:
The changes to the `build-plan` buildpack have been added and a generic `plan.toml` structure has been added to the RFC in the latest commit.